src/Stripe.net/Infrastructure/ParameterBuilder.cs

```diff
@@ -33,6 +33,16 @@ public static string ApplyAllParameters(this StripeService service, object obj,
                             RequestStringBuilder.ProcessPlugins(ref requestString, attribute, property, value, obj);
                     }
                 }
+
+                var field = obj.GetType().GetRuntimeField("ExtraParams");
+                if (field != null)
```

Comment:
This was added temporarily right? Should we assert if you option doesn't have that value so that you know you forgot to inherit from `StripeParamsBase`?

Reply:
Unless that breaks the reflection logic beyond saving, I think a better solution would be to change the `ApplyAllParameters` signature to receive a `StripeParamsBase` (or whatever name we choose for the parent class) instead of an `object`.

But to answer your question, yes, all params classes should inherit from `StripeParamsBase` and so there's no reason why `ExtraParams` would not be available.

Reply:
So what I meant is that if we remove the if, then we can catch cases where someone creates a new Options class but forgets to inherit from the Base class.

Reply:
@ob-stripe Now that all class derive from `StripeBaseOptions` should we remove the check? Or assert if `field == null` as it should not happen?

Reply:
Yeah, an assert seems reasonable 👍